phpBB/phpbb/storage/adapter/local.php

```diff
@@ -234,4 +252,82 @@ public function write_stream($path, $resource)
 			throw new exception('STORAGE_CANNOT_COPY_RESOURCE');
 		}
 	}
+
+	/**
+	 * Get file size.
+	 *
+	 * @param string	$path	The file
+	 *
+	 * @throws \phpbb\storage\exception\exception		When cannot get size
+	 *
+	 * @return array Properties
+	 */
+	public function file_size($path)
+	{
+		$size = filesize($this->root_path . $path);
+
+		if ($size === null)
+		{
+			throw new exception('STORAGE_CANNOT_GET_FILESIZE');
+		}
+
+		return ['size' => $size];
+	}
+
+	/**
+	 * Get file mimetype.
+	 *
+	 * @param string	$path	The file
+	 *
+	 * @return array	Properties
+	 */
+	public function file_mimetype($path)
+	{
+		return ['mimetype' => $this->mimetype_guesser->guess($this->root_path . $path)];
+	}
+
+	/**
+	 * Get image dimensions.
+	 *
+	 * @param string	$path	The file
+	 *
+	 * @return array	Properties
+	 */
+	protected function image_dimensions($path)
+	{
+		$size = $this->imagesize->getImageSize($this->root_path . $path);
+
+		// For not supported types like swf
+		if ($size === false)
+		{
+			$imsize = getimagesize($this->root_path . $path);
+			$size = ['width' => $imsize[0], 'height' => $imsize[1]];
+		}
+
+		return ['image_width' => $size['width'], 'image_height' => $size['height']];
+	}
+
+	/**
+	 * Get image width.
+	 *
+	 * @param string	$path	The file
+	 *
+	 * @return array	Properties
+	 */
+	public function file_image_width($path)
+	{
+		return $this->image_dimensions($path);
```

Comment:
No, this is right, but i think i will remove file_info, i dont like how it works